### PR TITLE
workload: disable regulation of workload regions by default

### PR DESCRIPTION
**Pull request by AndrewMeadows**

### CODE REVIEW

File: interface/src/workload/GameWorkload.cpp [hunk 1]
```diff
@@ -15,13 +15,28 @@
...
+    const float MIN_NUM_STEPS_DOWN = 250.0f;
+    const float MIN_NUM_STEPS_UP = 375.0f;
+
+    for (int32_t i = 0; i < workload::Region::NUM_VIEW_REGIONS; ++i) {
+        glm::vec2 minBackFront = MIN_BACK_FRONT_SCALE_FACTOR * regionBackFronts[i];
+        glm::vec2 maxBackFront = MAX_BACK_FRONT_SCALE_FACTOR * regionBackFronts[i];
+        glm::vec2 stepDown = (maxBackFront - minBackFront) / MIN_NUM_STEPS_DOWN;
+        glm::vec2 stepUp = (maxBackFront - minBackFront) / MIN_NUM_STEPS_UP;
+        regionRegulators[i] = Regulator(std::chrono::milliseconds(TIME_BUDGET_MSEC), minBackFront, maxBackFront, stepDown, stepUp);
+    }
```
Comment (AndrewMeadows) [inline interface/src/workload/GameWorkload.cpp]: This was an effort to make the configuration of the regulation algorithm a little easier to manage/understand.  The old `speedUp` and `speedDown` varables didn't have units of "speed" per se, but were really step-sizes-per-frame so I changed their names and compute their default values a little differently.  However, the values here were tuned to be somewhere near the ballpark to the old behavior, modulo the larger default region sizes.

File: interface/src/workload/GameWorkload.cpp [hunk 2]
```diff
@@ -86,6 +109,22 @@ void ControlViews::regulateViews(workload::Views& outViews, const workload::Timi
...
+    for (int32_t i = workload::Region::NUM_VIEW_REGIONS - 2; i >= 0; --i) {
+        int32_t j = i + 1;
+        regionBackFronts[i] = regionRegulators[i].clamp(glm::min(regionBackFronts[i], regionBackFronts[j] - MIN_REGION_GAP));
+    }
+    // enforce inside --> out
+    for (int32_t i = 1; i < workload::Region::NUM_VIEW_REGIONS; ++i) {
+        int32_t j = i - 1;
+        regionBackFronts[i] = regionRegulators[i].clamp(glm::max(regionBackFronts[i], regionBackFronts[j] + MIN_REGION_GAP));
+    }
+}
```
Comment (AndrewMeadows) [inline interface/src/workload/GameWorkload.cpp]: This `enforceRegionContainment()` method is new: it makes sure each inner region is inside its outer, and also enforces the final region limits.

File: interface/src/workload/GameWorkload.cpp [hunk 3]
```diff
@@ -41,37 +56,45 @@ void ControlViews::run(const workload::WorkloadContextPointer& runContext, const
         auto config = std::static_pointer_cast<Config>(runContext->jobConfig);
         config->dataExport = _dataExport;
         config->emitDirty();
+    } else {
+        for (auto& outView : outViews) {
+            outView.regionBackFronts[workload::Region::R1] = regionBackFronts[workload::Region::R1];
+            outView.regionBackFronts[workload::Region::R2] = regionBackFronts[workload::Region::R2];
+            outView.regionBackFronts[workload::Region::R3] = regionBackFronts[workload::Region::R3];
+            workload::View::updateRegionsFromBackFronts(outView);
+        }
```
Comment (AndrewMeadows) [inline interface/src/workload/GameWorkload.cpp]: When `_data.regulateViewRanges` is false we still need to export the values to the workload engine.

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/10069/HighFidelity-Beta-PR13226-f5fa49a1ab40d0d6deb120c30d0a9dc58f09e922.apk

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/10069/HighFidelity-Beta-Interface-PR13226-f5fa49a1ab40d0d6deb120c30d0a9dc58f09e922.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/10069/HighFidelity-Beta-PR13226-f5fa49a1ab40d0d6deb120c30d0a9dc58f09e922.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/10069/HighFidelity-Beta-Interface-PR13226-f5fa49a1ab40d0d6deb120c30d0a9dc58f09e922.exe

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/10117/HighFidelity-Beta-PR13226-cd568867b2d179c1a71b8d5170f2db3cf1ed52f9.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/10117/HighFidelity-Beta-PR13226-cd568867b2d179c1a71b8d5170f2db3cf1ed52f9.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/10117/HighFidelity-Beta-Interface-PR13226-cd568867b2d179c1a71b8d5170f2db3cf1ed52f9.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/10117/HighFidelity-Beta-PR13226-cd568867b2d179c1a71b8d5170f2db3cf1ed52f9.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/10117/HighFidelity-Beta-Interface-PR13226-cd568867b2d179c1a71b8d5170f2db3cf1ed52f9.exe

**Comment by AndrewMeadows** [comment]
**DO_NOT_MERGE**: something busted when I last rebased.  I need to track it down.

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/10185/HighFidelity-Beta-PR13226-57138b5fb742487e6a73984525a05ffa3801bd90.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/10185/HighFidelity-Beta-PR13226-57138b5fb742487e6a73984525a05ffa3801bd90.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/10185/HighFidelity-Beta-Interface-PR13226-57138b5fb742487e6a73984525a05ffa3801bd90.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/10185/HighFidelity-Beta-PR13226-57138b5fb742487e6a73984525a05ffa3801bd90.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/10185/HighFidelity-Beta-Interface-PR13226-57138b5fb742487e6a73984525a05ffa3801bd90.exe

**Pull request merged**